libkbfs/disk_limits_unix.go

```diff
@@ -0,0 +1,31 @@
+// Copyright 2017 Keybase Inc. All rights reserved.
+// Use of this source code is governed by a BSD
+// license that can be found in the LICENSE file.
+
+// +build !windows
+
+package libkbfs
+
+import (
+	"github.com/pkg/errors"
+	"golang.org/x/sys/unix"
+)
+
+// getDiskLimits gets the disk limits for the logical disk containing
+// the given path.
+//
+// TODO: Also return available files.
+func getDiskLimits(path string) (availableBytes uint64, err error) {
+	var stat unix.Statfs_t
+	err = unix.Statfs(path, &stat)
+	if err != nil {
+		return 0, errors.WithStack(err)
+	}
+
+	// Bavail is the free block count for an unprivileged user.
+	availableBytes = stat.Bavail * uint64(stat.Bsize)
+
+	// TODO: Use stat.Ffree to return availableFiles.
+
+	return availableBytes, nil
```

Comment:
I vote for available bytes, since using total bytes can lead to bad behavior with small disks, even when the disk isn't full. For example, a 10 GB disk that's almost full, say, only 2 GB free, would have all that remaining space gobbled up by the journal with a 25% limit.

I didn't think of the fact that available bytes already counts the journal space, but fortunately that's easy enough to fix -- if we want J <= p(J + A), then that's equivalent to requiring J <= (p/(1-p)) A; in this case, if we want p = 25%, then it suffices to require that the initial disk limit be 33 1/3% of the current available disk space. (This took me embarrassingly long to work out.) This is all approximate since we're only counting blocks, but we'll have to handle disk full errors separately.

What do you think?